### PR TITLE
[XCConfig Helper] Only link dynamic vendored frameworks and libraries

### PR DESCRIPTION
**Pull request by kcoleman731**
This pull request addresses issue [4336](https://github.com/CocoaPods/CocoaPods/issues/4336). 

CocoaPods will now only add dynamic vendored frameworks and libraries to target build settings for pod dependencies. It will not add static libraries and frameworks as it is unnecessary. 

cc @neonichu @segiddins @mhuusko5

### CODE REVIEW

File: spec/unit/generator/xcconfig/aggregate_xcconfig_spec.rb [hunk 1]
```diff
@@ -157,8 +157,8 @@ def spec
               fixture_spec('monkey/monkey.podspec')
             end
 
-            it 'does not add the framework build path to the xcconfig' do
-              @xcconfig.to_hash['FRAMEWORK_SEARCH_PATHS'].should.be.nil
+            it 'does add the framework build path to the xcconfig' do
+              @xcconfig.to_hash['FRAMEWORK_SEARCH_PATHS'].should.not.be.nil
             end
```
Comment (kcoleman731) [inline spec/unit/generator/xcconfig/aggregate_xcconfig_spec.rb]: `monkey.podspec` has been updated to reference a dynamic `vendored_framework`. As such, `FRAMEWORK_SEARCH_PATHS` should no longer be nil.

**Comment by segiddins** [comment]
Looks good!

File: lib/cocoapods/generator/xcconfig/xcconfig_helper.rb [hunk 2]
```diff
@@ -64,19 +65,36 @@ def self.add_settings_for_file_accessors_of_target(target, xcconfig)
         # @param [Xcodeproj::Config] xcconfig
         #        The xcconfig to edit.
         #
-        def self.add_vendored_dependency_build_settings(target, xcconfig)
+        def self.add_static_dependency_build_settings(target, xcconfig, file_accessor)
```
Comment (segiddins) [inline lib/cocoapods/generator/xcconfig/xcconfig_helper.rb]: needs docs for the new argument

Comment (kcoleman731) [inline lib/cocoapods/generator/xcconfig/xcconfig_helper.rb]: :+1:

**Comment by segiddins** [comment]
:+1:

**Comment by neonichu** [comment]
Great work, @kcoleman731 :tada:

**Comment by segiddins** [comment]
Ah crap, we forgot a CHANGELOG entry. @kcoleman731 want to send along another PR adding one?

**Pull request merged**